domain/consensus/processes/headersselectedtipmanager/headersselectedtipmanager_test.go

```diff
@@ -0,0 +1,86 @@
+package headersselectedtipmanager_test
+
+import (
+	"github.com/kaspanet/kaspad/domain/consensus"
+	"github.com/kaspanet/kaspad/domain/consensus/model/externalapi"
+	"github.com/kaspanet/kaspad/domain/consensus/utils/testutils"
+	"github.com/kaspanet/kaspad/domain/dagconfig"
+	"testing"
+)
+
+func TestAddHeaderTip(t *testing.T) {
+	testutils.ForAllNets(t, true, func(t *testing.T, params *dagconfig.Params) {
+		factory := consensus.NewFactory()
+		tc, tearDown, err := factory.NewTestConsensus(params, false, "TestAddHeaderTip")
+		if err != nil {
+			t.Fatalf("NewTestConsensus: %s", err)
+		}
+		defer tearDown(false)
+
+		checkExpectedSelectedChain := func(expectedSelectedChain []*externalapi.DomainHash) {
+			for i, blockHash := range expectedSelectedChain {
+				chainBlockHash, exists, err := tc.HeadersSelectedChainStore().GetHashByIndex(tc.DatabaseContext(), uint64(i))
+				if err != nil {
+					t.Fatalf("GetHashByIndex: %+v", err)
+				}
+
+				if !exists {
+					t.Fatalf("index %d is expected to exist", i)
+				}
+
+				if !blockHash.Equal(chainBlockHash) {
+					t.Fatalf("chain block %d is expected to be %s but got %s", i, blockHash, chainBlockHash)
+				}
+
+				index, exists, err := tc.HeadersSelectedChainStore().GetIndexByHash(tc.DatabaseContext(), blockHash)
+				if err != nil {
+					t.Fatalf("GetIndexByHash: %+v", err)
+				}
+
+				if !exists {
+					t.Fatalf("hash %s is expected to exist", blockHash)
+				}
+
+				if uint64(i) != index {
+					t.Fatalf("chain block %s is expected to be %d but got %d", blockHash, i, index)
+				}
+			}
+
+			_, exists, err := tc.HeadersSelectedChainStore().GetHashByIndex(tc.DatabaseContext(),
+				uint64(len(expectedSelectedChain)+1))
+			if err != nil {
+				t.Fatalf("GetIndexByHash: %+v", err)
+			}
+
+			if exists {
+				t.Fatalf("index %d is not expected to exist", uint64(len(expectedSelectedChain)+1))
+			}
```

Comment:
Please make sure than the store's highest index is correct as well.

Reply:
The highest index is an implementation detail that helps to maintain the selected chain and is not exposed by the store, so I think it's enough to check if the selected chain is properly maintained.